src/test/integration/bearer.test.ts

```diff
@@ -0,0 +1,107 @@
+import express from 'express';
+import got from 'got';
+import {
+    Authenticator,
+    AuthistOptions,
+    createAuthenticator,
+    ErrorCode,
+    saveNonExistingUser,
+    signInWithEmailAndPassword,
+    User,
+} from '../../lib';
+import { getUserByEmail, getUserById, initDb } from '../databaseUtils';
+
+let userModel: any;
+const serverUrl = 'http://localhost:3000';
+const createServer = (authenticator: Authenticator) => {
+    const app = express();
+    app.get('/', authenticator.expressBearer, (req: any, res) => res.json(req.user));
+    return app;
+};
+const getAuthenticator = (options: AuthistOptions) => createAuthenticator(options);
+
+describe('Bearer authentication middleware', () => {
+    const getAuthistOptions = (): AuthistOptions => ({
+        getUserById: getUserById(userModel),
+        emailPassword: {
+            getUserByEmail: getUserByEmail(userModel),
+            saveNonExistingUser: (data, hashedPassword, _req) =>
+                userModel.create({ email: data.email, password: hashedPassword }),
+        },
+    });
+    const email = 'tester@test.com';
+    const password = '***';
+    beforeAll(async () => {
+        userModel = await initDb();
+        const authenticator = getAuthenticator(getAuthistOptions());
+        const server = createServer(authenticator);
+        server.listen(3000);
```

Comment:
(optional) There is a better way to do this than using specific port - use port: 0. This way the server starts on any available port the OS has to offer. There should be a way to access it from express too, it's `server.address().port` on the net.Server instance if you are using native.

https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback


The huge benefit besides the hardcoded port (and failing the tests when your OS is using that port) is that you can easily start multiple servers during the tests in parallel, without worrying about port clash.

Reply:
I used port 0

Reply:
Looks good